File: blog-sever/server-java/src/main/java/com/site/blog/service/AdminUserService.java
package com.site.blog.service;

import com.site.blog.model.entity.AdminUser;
import com.baomidou.mybatisplus.extension.service.IService;

/**
 * <p>
 * 后台管理员信息表 服务类
 * </p>
 *
 */
public interface AdminUserService extends IService<AdminUser> {

    boolean validatePassword(String userId, String oldPwd);

    boolean updateUserInfo(AdminUser adminUser);

    int register(AdminUser user);

    /**
     * 根据adminuserid获取用户
     * @param id
     * @return
     */
    AdminUser getAdminUserById(String id);
}

File: myblog-jpa/src/main/kotlin/com/site/blog/service/AdminUserService.kt
package com.site.blog.service


import com.site.blog.model.entity.AdminUser

/**
 *
 *
 * 后台管理员信息表 服务类
 *
 *
 */
interface AdminUserService  {
    fun validatePassword(userId: Long, oldPwd: String): Boolean
    fun updateUserInfo(adminUser: AdminUser): Boolean
    fun register(user: AdminUser): Int

    /**
     * 根据adminuserid获取用户
     * @param id
     * @return
     */
    fun getAdminUserById(id: String): AdminUser
}